src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -198,6 +198,12 @@ export class AddScenarioModal implements OnInit, OnDestroy {
         this.newScenForm.get('region').valueChanges.subscribe(item => {
              if(item != null){
                 this.selectedRegion = item;
```

Comment:
If we change this to `this._nssService.setSelectedRegion(item);`, it'll update the regression regions _and_ the selected region, which means we won't need lines 202-205